artiq/coredevice/exceptions.py

```diff
@@ -77,7 +77,10 @@ def single_traceback(self, exception_index):
             exn_id = int(exn_id)
         else:
             exn_id = 0
-        lines.append("{}({}): {}".format(name, exn_id, message.format(*params)))
+        try:
+            lines.append("{}({}): {}".format(name, exn_id, message.format(*params)))
+        except:
+            lines.append("{}({}): {}".format(name, exn_id, message))
```

Comment:
It is not a failure, if user will have some message which would break the `message.format(*params)`, it will just not be formatted, which is intended behavior. User is not able to use such messages anyway

Reply:
Because the alternative is to add marker on the compile stage, and then transmit it each time with exception message. Otherwise it all will be hackable one way or another

Reply:
And also changing the protocol and compiler seems to be overkill, since we do not assume it to be secure